### ISSUE TITLE
Mask2former Support.

### ISSUE THREAD

**Issue by Hulmes0217** [rankseg/rankseg#7]
rankseg:
<img width="290" height="161" alt="Image" src="https://github.com/user-attachments/assets/7d7d4d72-856e-4cbe-b883-563b7738e98b" />

argmax:
<img width="315" height="159" alt="Image" src="https://github.com/user-attachments/assets/b1bd9ddf-e060-4b45-924f-0b851f6978c5" />

Thanks for this wonderful work. I tried to apply it to Mask2former with MMSeg 1.x version, but I found the rankseg decoder got a worse result. Could you help me with this problem?

**Comment by statmlben** [comment]
Thank you for the comments. RankSEG relies on the outcome of pixel-wise logits/probs to find a better segmentation mask under Dice/IoU metrics.

Mask2Former operates on a 'Mask Classification' paradigm, where the reconstructed logits/probs are often over confident and sparse NOT real pixel-wise logits/probs. Therefore, RankSEG is best suited for pixel-wise prediction models (e.g., DeepLab, SegFormer, UNet) rather than mask-based transformers.

**Comment by ZixunWang** [comment]
@Hulmes0217 Thanks for your interest and for raising this question! 

As @statmlben mentioned, RankSEG requires well-estimated probability masks as input, which is typically the case for traditional segmentation models trained with cross-entropy loss.

Here are some reasons that come to mind for this case:
1. The training process of Mask2Former involves mask generation and bipartite matching, which deviates from likelihood estimation and thus makes the final output difficult to interpret as probabilities.
2. The loss function that combines soft Dice/IoU also makes the model output deviate from calibrated probabilities.

In short, Mask2Former-like models essentially output some "good confidence scores" rather than "good probabilities".
